### PR TITLE
Fix an issue that caused us to not get compiler arguments when opening a SwiftPM package at a symlink

### PR DESCRIPTION
**Pull request by ahoppen**
When you had a package at `/pkg_real` and a symlink `/pkg` pointing to `/pkg_real`, then opened a workspace at `/pkg`, we wouldn’t get any build settings for any of the files. This was masked in tests because they still called `SwiftPMBuildSystem.targets(for:)`, which handled symlink resolution but wasn’t called in production.

Handle symlink resolution `BuildSystemManager`, remove `SwiftPMBuildSystem.targets(for:)` and its related members/methods, and migrate the tests to go through `BuildSystemManager`, which is what production code does. A nice side effect of this is that the tests will log the requests sent to the build system.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Sources/BuildSystemIntegration/BuildSystemManager.swift [hunk 1]
```diff
@@ -78,6 +79,18 @@ fileprivate extension BuildTarget {
   }
 }
 
+fileprivate extension DocumentURI {
+  /// If this is a file URI pointing to a symlink, return the realpath of the URI, otherwise return `nil`.
+  var symlinksResolved: DocumentURI? {
```
Comment (hamishknight) [inline Sources/BuildSystemIntegration/BuildSystemManager.swift:84]: IMO `symlinksResolved` doesn't really communicate "returns `nil` if not a symlink", maybe it should be something more like `symlinkTarget`?

**Review by hamishknight — APPROVED** [review]

File: Sources/BuildSystemIntegration/BuildSystemManager.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/BuildSystemIntegration/BuildSystemManager.swift:84]: Good idea 👍🏽

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**